public/app/core/components/GraphNG/GraphNG.tsx

```diff
@@ -139,6 +139,11 @@ export class GraphNG extends Component<GraphNGProps, GraphNGState> {
   }
 
   handleCursorUpdate(evt: DataHoverEvent | LegacyGraphHoverEvent) {
+    // ignore uplot-emitted events, since we already use uPlot's sync
+    if (evt.tags?.has('uplot')) {
```

Comment:
guess we could have `hasTag[s](...)` method alongside `setTags` to avoid `?` when checking for tag existence. And then tags become private property on the event class.

Reply:
this remove ability to enumerate them freely without adding more methods. the hasTag version requires that you know exactly what you're looking for and cannot do something like checking for namespaced tags like `uplot.foo` and filtering on startsWith `uplot`.

but happy to change it, too.